File: python/accel_purge.py
import xmlrpc.client

API_ADDRESS = '192.168.134.135:8080'
API_KEY = ''
API_SECRET = ''
API_MODULE = 'snaptNginx'

# the URL is always /api/apikey/apisecret/module - in this case module is snaptNginx for the Accelerator
API_FULL_URL = f'http://{API_ADDRESS}/api/{API_KEY}/{API_SECRET}/{API_MODULE}'


def list_methods(proxy):
    methods = proxy.system.listMethods()
    methods.sort()
    return methods


def purge_cache_string(proxy, purge_string):
    return proxy.snaptNginx.purgeCacheString(purge_string)


with xmlrpc.client.ServerProxy(API_FULL_URL) as proxy:
    methods = list_methods(proxy)
    for method in methods:
        if method.startswith('system'):
            continue
        print(f'- {method}')

    purge_string = 'snapt-ui.com'

    purge_cache_string_result = purge_cache_string(proxy, purge_string)
    print(f'Purge cache string: {purge_cache_string_result}')

File: ruby/accel_purge.rb
require "xmlrpc/client"

API_ADDRESS = "192.168.1.20:8080"
API_KEY = "1f29-1266-694-18ec"
API_SECRET = "SNP5f8d8ef59d229240105044"
API_MODULE = "snaptNginx"

# the URL is always /api/apikey/apisecret/module - in this case module is snaptHA for the Balancer
API_FULL_URL = "http://#{API_ADDRESS}/api/#{API_KEY}/#{API_SECRET}/#{API_MODULE}"

def list_methods(client)
  client.call("system.listMethods")
end

def purge_cache_string(client, purge_string)
  client.call("#{API_MODULE}.purgeCacheString", purge_string)
end

client = XMLRPC::Client.new2(API_FULL_URL)

methods = list_methods(client)
methods.each do |method|
  if method.start_with? "system"
    next
  end
  puts "- #{method}"
end

purge_cache_string_result = purge_cache_string(client, "snapt-ui.com")
puts "Purge cache string: #{purge_cache_string_result}"
